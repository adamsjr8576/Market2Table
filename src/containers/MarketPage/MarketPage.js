import React from 'react';
import './MarketPage.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import images from '../../images/images.js';
import { toggleFavorite, controlFavorites } from '../../actions/index.js';
import MapContainer from '../MapContainer/MapContainer.js';

export const MarketPage = ({ id, markets, favorites, toggleFavorite, controlFavorites }) => {
  let market = markets.find(market => market.id === id);
  if (market === undefined) {
    market = favorites.find(market => market.id === id);
  }
  let image;
  market.favorite? image = images.redChili : image = images.chili;
  return (
    <main className="market-page-main">
      <section className="market-page-section">
        <div className="market-page-header-container">
          <h1 className="market-page-name">{market.marketname}</h1>
          <button className="favorites-button" type="button" onClick={() => {controlFavorites(market); toggleFavorite(id)}}><img src={image} alt="chili icon" className="chili-icon" /></button>
        </div>
        <div className="market-page-info-container">
          <h2 className="market-page-header">Address:</h2>
          <p className="products-p">{market.Address}</p>
        </div>
        <div className="market-page-info-container">
          <h2 className="market-page-header">Products:</h2>
          <p className="products-p">{market.Products}</p>
        </div>
        <div className="market-page-info-container">
          <h2 className="market-page-header">Schedule:</h2>
          <p className="products-p">{market.Schedule}</p>
        </div>
        <a href={market.GoogleLink}>Google Maps</a>
      </section>
      <section className="markets-map-section">
        <MapContainer market={market}/>
      </section>
    </main>
  )
}

export const mapDispatchToProps = dispatch => ({
  toggleFavorite: (id) => dispatch( toggleFavorite(id) ),
  controlFavorites: market => dispatch( controlFavorites(market) )
});

export const mapStateToProps = state => ({
  markets: state.farmersMarkets,
  favorites: state.favorites
});

export default connect(mapStateToProps, mapDispatchToProps)(MarketPage);

MarketPage.propTypes = {
  id: PropTypes.string,
  markets: PropTypes.arrayOf(PropTypes.object),
  favorites: PropTypes.array,
  toggleFavorite: PropTypes.func,
  controlFavorites: PropTypes.func
}
