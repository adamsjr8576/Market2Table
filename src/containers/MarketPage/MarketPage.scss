.market-page-main {
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: 85vh;
}

.market-page-section {
  height: 75vh;
  border: 3px solid black;
  background-color: #FFE7BF;
  width: 550px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
}

.market-page-header-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
}

.market-page-header {
  font-size: 2em;
  text-decoration: underline;

}
