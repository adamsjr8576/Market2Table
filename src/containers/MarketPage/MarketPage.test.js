import React from 'react';
import { shallow } from 'enzyme';
import { MarketPage, mapDispatchToProps, mapStateToProps } from './MarketPage';
import { toggleFavorite, controlFavorites } from '../../actions/index.js';

describe('MarketPage', () => {
  describe('MarketPage Component', () => {
    let mockMarkets;
    let mockId;
    let mockFavorites;
    let mockToggleFavorite;
    let mockControlFavorites;

    beforeEach(() => {
      mockMarkets = [
         {
           id: "1002192",
           marketname: "Ballston FRESHFARM Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         },
         {
           id: "22008839",
           marketname: "Ballston Farmers Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         }];
        mockFavorites = [
          {
          id: "39",
          marketname: "Golden Farmers Market",
          favorite: false,
          GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          Address: "here",
          Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
          Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          latitude: "38.881112",
          longitude: "-77.112179"
          }
        ]
        mockToggleFavorite = jest.fn();
        mockControlFavorites = jest.fn();
    });

    it('Should match the snapshot with all the right info passed in', () => {
      mockId = "22008839";
      const wrapper = shallow(<MarketPage
        id={mockId}
        markets={mockMarkets}
        favorites={mockFavorites}
        toggleFavorite={mockToggleFavorite}
        controlFavorites={mockControlFavorites}
        />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Should match the snapshot with all the right info passed in', () => {
      mockId = "39";
      const wrapper = shallow(<MarketPage
        id={mockId}
        markets={mockMarkets}
        favorites={mockFavorites}
        toggleFavorite={mockToggleFavorite}
        controlFavorites={mockControlFavorites}
        />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should invoke toggleFavorite and controlFavorites when button is clicked', () => {
      mockId = "22008839";
      const mockMarket = {
        id: "22008839",
        marketname: "Ballston Farmers Market",
        favorite: false,
        GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
        Address: "901 N Taylor St, Ballston, Virginia, 22203",
        Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
        Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
        latitude: "38.881112",
        longitude: "-77.112179"
      }
      const wrapper = shallow(<MarketPage
        id={mockId}
        markets={mockMarkets}
        favorites={mockFavorites}
        toggleFavorite={mockToggleFavorite}
        controlFavorites={mockControlFavorites}
        />);

      wrapper.find('.favorites-button').simulate('click');

      expect(mockToggleFavorite).toHaveBeenCalledWith(mockId);
      expect(mockControlFavorites).toHaveBeenCalledWith(mockMarket);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockMarket;
    let mockDispatch;

    beforeEach(() => {
      mockMarket = {
        id: "22008839",
        marketname: "Ballston Farmers Market",
        favorite: false,
        GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
        Address: "901 N Taylor St, Ballston, Virginia, 22203",
        Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
        Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
        latitude: "38.881112",
        longitude: "-77.112179"
      }
      mockDispatch = jest.fn();
    });

    it('should call dispatch with toggleFavorite', () => {
      const actionToDispatch = toggleFavorite(mockMarket.id);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.toggleFavorite(mockMarket.id);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with controlFavorites', () => {
      const actionToDispatch = controlFavorites(mockMarket);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.controlFavorites(mockMarket);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with markes and favorites', () => {
      const mockState = {
        farmersMarkets:
        [
          {
            id: "1002192",
            marketname: "Ballston FRESHFARM Market",
            favorite: false,
            GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
            Address: "901 N Taylor St, Ballston, Virginia, 22203",
            Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
            Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
            latitude: "38.881112",
            longitude: "-77.112179"
          },
          {
            id: "22008839",
            marketname: "Ballston Farmers Market",
            favorite: false,
            GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
            Address: "901 N Taylor St, Ballston, Virginia, 22203",
            Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
            Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
            latitude: "38.881112",
            longitude: "-77.112179"
          }],
        favorites: [],
        zipCode: '80401'
      }
      const expected = {
        markets:
        [
          {
            id: "1002192",
            marketname: "Ballston FRESHFARM Market",
            favorite: false,
            GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
            Address: "901 N Taylor St, Ballston, Virginia, 22203",
            Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
            Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
            latitude: "38.881112",
            longitude: "-77.112179"
          },
          {
            id: "22008839",
            marketname: "Ballston Farmers Market",
            favorite: false,
            GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
            Address: "901 N Taylor St, Ballston, Virginia, 22203",
            Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
            Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
            latitude: "38.881112",
            longitude: "-77.112179"
          }],
        favorites: [],
      }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
