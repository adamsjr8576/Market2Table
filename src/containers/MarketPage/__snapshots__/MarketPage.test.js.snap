// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketPage MarketPage Component Should match the snapshot with all the right info passed in 1`] = `
<main
  className="market-page-main"
>
  <section
    className="market-page-section"
  >
    <div
      className="market-page-header-container"
    >
      <h1
        className="market-page-name"
      >
        Ballston Farmers Market
      </h1>
      <button
        className="favorites-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="chili icon"
          className="chili-icon"
          src="chilli.png"
        />
      </button>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Address:
      </h2>
      <p
        className="products-p"
      >
        901 N Taylor St, Ballston, Virginia, 22203
      </p>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Products:
      </h2>
      <p
        className="products-p"
      >
        Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry
      </p>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Schedule:
      </h2>
      <p
        className="products-p"
      >
        June - October Thursday 3:00 PM to 7:00 PM
      </p>
    </div>
    <a
      href="http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)"
    >
      Google Maps
    </a>
  </section>
  <section
    className="markets-map-section"
  >
    <Wrapper
      market={
        Object {
          "Address": "901 N Taylor St, Ballston, Virginia, 22203",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "22008839",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Ballston Farmers Market",
        }
      }
    />
  </section>
</main>
`;

exports[`MarketPage MarketPage Component Should match the snapshot with all the right info passed in 2`] = `
<main
  className="market-page-main"
>
  <section
    className="market-page-section"
  >
    <div
      className="market-page-header-container"
    >
      <h1
        className="market-page-name"
      >
        Golden Farmers Market
      </h1>
      <button
        className="favorites-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="chili icon"
          className="chili-icon"
          src="chilli.png"
        />
      </button>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Address:
      </h2>
      <p
        className="products-p"
      >
        here
      </p>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Products:
      </h2>
      <p
        className="products-p"
      >
        Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry
      </p>
    </div>
    <div
      className="market-page-info-container"
    >
      <h2
        className="market-page-header"
      >
        Schedule:
      </h2>
      <p
        className="products-p"
      >
        June - October Thursday 3:00 PM to 7:00 PM
      </p>
    </div>
    <a
      href="http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)"
    >
      Google Maps
    </a>
  </section>
  <section
    className="markets-map-section"
  >
    <Wrapper
      market={
        Object {
          "Address": "here",
          "GoogleLink": "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          "Products": "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          "Schedule": "June - October Thursday 3:00 PM to 7:00 PM",
          "favorite": false,
          "id": "39",
          "latitude": "38.881112",
          "longitude": "-77.112179",
          "marketname": "Golden Farmers Market",
        }
      }
    />
  </section>
</main>
`;
